HTWDD/Components/Canteen/Models/Canteen.swift

```diff
@@ -8,87 +8,47 @@
 
 import Foundation
 import CoreLocation
-import OpenMensaKit
 import RxSwift
 
 struct Canteen {
+    let name: String
+    let id: Id
+    let coordinate: CLLocationCoordinate2D?
 
-    enum Id: Int {
-        case reichenbachstrasse = 80
+    static let reichenbachstrasse = Canteen(name: "Mensa Reichenbachstraße",
+                                            id: .reichenbachstrasse,
+                                            coordinate: CLLocationCoordinate2D(latitude: 51.0342243,
+                                                                               longitude: 13.7318803))
+    static let all = [reichenbachstrasse]
 
-        var imageId: Int? {
-            switch self {
-            case .reichenbachstrasse: return 9
-            }
+    static func with(id: Id) throws -> Canteen {
+        guard let canteen = all.filter({ $0.id == id }).first else {
+            throw Canteen.Error.incompatibleCanteen(id: id)
         }
+        return canteen
     }
 
-    let id: Id
-    let name: String
-    let address: String
-    let coordinate: CLLocationCoordinate2D?
-
-    private let canteen: OpenMensaKit.Canteen
+    enum Id: String {
+        case reichenbachstrasse = "mensa_reichenbachstraße"
 
-    init?(_ openMensaCanteen: OpenMensaKit.Canteen) {
-        guard let id = Id(rawValue: openMensaCanteen.id) else {
-            return nil
+        var imageId: Int? {
```

Comment:
Why Optional? Currently there is only an Int returned :)

Reply:
I wish the compiler would throw warnings on unnecessary optionals.